File: algorithm/plot_csv.m
% Import data from CSV file
data = readmatrix('correction_Param.csv');

% Extract columns
X = data(:,4:6);
Y = data(:,end);

% Plot data
scatter3(X(:,1), X(:,2), Y);
xlabel('azimuth');
ylabel('elevation');
zlabel('dis');

%{
% Generate some sample data
x = gpuArray(linspace(0, 1, 100)');
y = gpuArray(sin(2*pi*x) + 0.1*randn(size(x)));

% Create a feedforward neural network with one hidden layer
net = feedforwardnet(10);

% Set the training options
options = trainingOptions('sgdm', ...
    'MaxEpochs', 100, ...
    'MiniBatchSize', 64, ...
    'Shuffle', 'every-epoch', ...
    'Verbose', false, ...
    'Plots', 'training-progress', ...
    'UseGPU', true); % enable GPU acceleration

% Train the network
net = train(net, x, y, options);    
%}

File: algorithm/plot_csv.py
import pandas as pd
import matplotlib.pyplot as plt
import os

base_path = os.path.dirname(__file__)

# Load the data from the CSV file
data = pd.read_csv(os.path.join(base_path, 'correction_Param.csv'))

# Extract the feature values and target values
x1 = data['azimuth']
x2 = data['elevation']
y = data['correction_param']

# Create a scatter plot to visualize the relationship between x1, x2, and y
plt.figure(figsize=(8, 6))
plt.scatter(x1, y, label='azimuth')
plt.scatter(x2, y, label='elevation')
plt.xlabel('azimuth and elevation')
plt.ylabel('y')
plt.legend()
plt.title('Relationship between azimuth, elevation, and y')
plt.show()
